frontend-spa/src/js/newartistform.js

```diff
@@ -0,0 +1,55 @@
+import{
+    DomMaker
+    } from './DomMaker.js'
+    import{
+        InputMaker
+    } from './InputMaker.js'
+    import{getAllArtists} from './artistbuilder.js'
+    
+    const createNewArtistForm= ()=>{
+        const form = new DomMaker('form')
+                        .changeContent('Add new Artist:')
+                        .appendChild(new DomMaker('Name')
+                                        .changeContent('Name:')
+                                        .render())
+                        .appendChild(new DomMaker('Age')
+                                        .changeContent('Age:')
+                                        .render())
+                        .appendChild(new DomMaker('Label')
+                                        .addClassName('form__description')
+                                        .render())
+                        .appendChild(new DomMaker('Albums')
+                                        .addClassName('form__description')
+                                        .render())
+                        .appendChild(new DomMaker('Songs')
+                                        .addClassName('form__description')
+                                        .render())
+                        .appendChild(new DomMaker('button')
+                                        .changeContent('Submit New Artist')
+                                        .addOnClickAction(()=>{
+                                         collectData();
+                                    })
+
+                                    .render())
+        return form;
+    }
+    
+    const collectData = () =>{
+        const newName = document.querySelector('.form__name');
+        const newDescription = document.querySelector('.form__description');
+    
+        const newArtistJson = {
+            "name": newArtist.value,
+            "description": newArtist.value
+        }
```

Comment:
Where is `newArtist` defined?
```suggestion
        const newArtistJson = {
            "name": newName,
            "description": newDescription
        }
```
The two variables above this, `newName` and `newDescription` are unused otherwise